app/app/calculations/punching.py

```diff
@@ -0,0 +1,212 @@
+                    # dla fundamentow mozna zredukowac sile do VEdred - por. 6.4.4(2)
+
+    # ############## OBLICZENIA ##############
+    # -----------------------------------------
+    # MATERIAL
+    fcd = alfa_cc * fck / gamma_c
+    fyd = fyk / gamma_s
+    fywd = alfa_sw * fyk
+
+    # -----------------------------------------
+    # GEOMETRY
+    d = (dx + dy) / 2 - ad  # spr. czy sie nie koliduje z 'D' (srednica kola)
+
+    # TODO(LZ): Dodac warunek na section i powiazac support
+
+    # TODO: ściana - naroże (3) i ściana - koniec (4)
+    support_id = support.id
+    if support_id == 0:     # słup wewnętrzny
+        u00 = 2 * (cx + cy)
+    elif support_id == 1:   # słup krawędziowy X
+        u00 = min(2 * cx + cy, cy + 3 * d)
+    elif support_id == 2:   # słup krawędziowy Y
+        u00 = min(cx + 2 * cy, cx + 3 * d)
+    elif support_id == 3:   # słup narożny
+        u00 = min(cx + cy, 3 * d)
+    elif support_id == 4:   # ściana - naroże
+        u00 = 100           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-naroże' niedostępna.")
+    elif support_id == 5:   # ściana - koniec
+        u00 = 200           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-koniec' niedostępna.")
+    else:
+        # TODO(LZ): dodac sciany (naroznik, koniec)
+        ERRORS.append('Błąd w określeniu położenia podpory (u0).')
+        u00 = 0.0
+    u0 = (1.0 - lambda_u / 100) * u00
+
+    if support_id == 0:     # slup wewnetrzny
+        u11 = 2 * cx + 2 * cy + 4 * math.pi * d
+    elif support_id == 1:   # słup krawędziowy X
+        u11 = 2 * cx + 1 * cy + 2 * math.pi * d + 2 * lx
+    elif support_id == 2:   # słup krawędziowy Y
+        u11 =  1 * cx + 2 * cy + 2 * math.pi * d + 2 * ly
+    elif support_id == 3:   # słup narożny
+        u11 = 1 * cx + 1 * cy + 1 * math.pi * d + lx + ly
+    elif support_id == 4:   # ściana - naroże
+        u11 = 100           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-naroże' niedostępna.")
+    elif support_id == 5:   # ściana - koniec
+        u11 = 200           # TODO(LZ): poprawic!
+        ERRORS.append("Podpora 'ściana-koniec' niedostępna.")
+    else:
+        # TODO(LZ): dodac sciany (naroznik, koniec)
+        ERRORS.append('Błąd w określeniu położenia podpory (u1).')
+        u11 = 0.0
+    u1 = (1.0 - lambda_u / 100) * u11
+
+    # -----------------------------------------
+    # NAPREZENIA
+    # Maksymalne naprezenie przy przebiciu (vEd)
+    ved0 = beta * ved * (u0 * d)
+    ved1 = beta * ved * (u1 * d)
+
+    # Maksymalna wytrzymalosc na ścinanie przy przebiciu ze wzgledu na sciskane krzyzulce betonowe (vRdmax)
+    ni = 0.6 * (1.0 - fck / (250 * MPa))
+    vrdmax = 0.4 * ni * fcd
+
+    if vrdmax >= ved0:
+        ERRORS.append('Nośność betonu na ścinanie przy przebiciu zachowana (vEd0 <= vRdmax).')
+    else:   # elif vrdmax < ved0:
+        ERRORS.append('Nośność betonu na ścinanie przekroczona (vEd0 > vRdmax)!')
```

Comment:
Te dwa komunikaty nie mają wydźwięku negatywnego i nie wskazują na błąd, a jednak są błędem. A! Jedno nie jest błędem, a drugie jest. Jeśli chcemy zachować pewne info dla usera, jak jest wszystko ok, to proponuję stworzyć nową listę INFO albo coś takiego i tam wrzucać. Czyli pierwsza opcja doda do INFO, a druga do ERRORS. Niżej też to występuje. Bo na tym etapie zadania, po kliknięciu dostałam dwa błędy, ale z komuniaktów nie wynika, że coś jest źle :]